src/services/mongoose/users.ts

```diff
@@ -27,7 +27,28 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const findByUsername = async (username: Username): Promise<User | RepositoryError> => {
+    try {
+      const result = await User.findOne({ username })
```

Comment:
I wasn't sure if to implement this lookup like this or as `user.getUserByUsername(username)` since that schema method can still throw and we haven't started refactoring those as yet

Reply:
I like re-implementing it and not using the method that can throw... but you should also bring in the validation logic to be consistent:
```
  if (!username.match(regexUsername)) {
    return null
  }

  const user = await this.findOne({ username: caseInsensitiveRegex(username) })
  ```
  
  In perticular that the lookup is case insensitive is important here.

Reply:
Agreed on the validation logic, implemented here now: https://github.com/GaloyMoney/galoy/pull/449/commits/7fa8be530da9f637479a70c1d660f2fafa88038c

To clarify, on the "replace + delete" do you mean instantiating a UserRepository and calling the new `findByUsername` method on it in all these places? :point_down: 

![image](https://user-images.githubusercontent.com/17693119/129890637-507ac3a6-49ff-4fa3-898d-e835811d64fb.png)

Reply:
Yeah we should avoid having duplicate login so I think you should replace all those calls

Reply:
Implementing here: https://github.com/GaloyMoney/galoy/compare/swap-in-new-add-invoice...swap-in-find-by-username